utils/utils.js

```diff
@@ -1187,3 +1187,30 @@ export function enforceWhitespace(el, offset, direction, rule) {
         spaceNode.nodeValue = spaceNode.nodeValue.replace(expr, spaceVisibility ? '\u00A0' : '');
     }
 }
+/**
+ * Generates an iterable of every node of a range.
+ * 
+ * @param {Range} range 
+ */
+export function* getRangeRover(range) {
+    const iterator = document.createNodeIterator(
+        range.commonAncestorContainer,
+        NodeFilter.SHOW_ALL, // pre-filter
+        {
+            // custom filter
+            acceptNode: function (node) {
+                return NodeFilter.FILTER_ACCEPT;
+            }
+        }
+    );
+    let started = false;
+    while (iterator.nextNode()) {
+        if (!started && iterator.referenceNode !== range.startContainer) { continue; }
+        started = true;
+        yield iterator.referenceNode;
+
+        if (iterator.referenceNode === range.endContainer) {
+            break;
+        }
+    }
+}
```

Comment:
Du coup la version réécrite complète avec tous les commentaires pris en compte ça donnerait:
```js
const iterator = document.createNodeIterator(range.startContainer);
let currentNode;
while ((currentNode = iterator.nextNode()) && currentNode !== range.endContainer) {
    yield currentNode;
}
```

Reply:
Ça fait pas la même chose, là tu itères juste sur le contenu de startContainer, qui contient pas forcément toute la sélection.

Reply:
En effet j'ai lu la doc un peu trop vite. C'est un peu chiant comme API quand même... Quand tu veux juste itérer d'un noeud A à un noeud B c'est juste pas pratique quoi...